apps/language_server/lib/language_server/providers/hover.ex

```diff
@@ -44,14 +44,174 @@ defmodule ElixirLS.LanguageServer.Providers.Hover do
     end)
   end
 
-  defp contents(%{docs: "No documentation available\n"}) do
+  defp contents(%{docs: "No documentation available\n"}, _subject \\ "") do
     []
   end
 
-  defp contents(%{docs: markdown}) do
+  defp contents(%{docs: markdown}, subject) do
     %{
       kind: "markdown",
-      value: markdown
+      value: add_hexdocs_link(markdown, subject)
     }
   end
+
+  defp add_hexdocs_link(markdown, subject) do
+    IO.inspect(subject)
+
+    [hd | tail] = markdown |> String.split("\n\n")
+
+    link = hexdocs_link(markdown, subject)
+
+    case link do
+      "" ->
+        markdown
+
+      _ ->
+        hd <> "  [view on hexdocs](#{link})\n\n" <> Enum.join(tail, "")
+    end
+  end
+
+  defp hexdocs_link(markdown, subject) do
+    t = markdown |> String.split("\n\n") |> hd() |> String.replace(">", "") |> String.trim()
+
+    cond do
+      func?(t) ->
+        mod_name = module_name(t)
+
+        if elixir_module?(mod_name) do
```

Comment:
I think we can go even further.
With `mod_name.__info__(:compile)` we can determine wether the module is from a dependency or not. Therefore, generate hexdocs for modules from dependencies as well

Reply:
yeah, i have fixed it.